### ISSUE TITLE
When selecting a date filter till "Today", charts should show represent that in the x-axis

### ISSUE THREAD

**Issue by paoliniluis** [metabase/metabase#16153]
Currently, Metabase only shows data points where there is data, no matter the selection in the time filters. So in the case that a user uses a filter like "<timestamp field> BETWEEN <some date> AND today()", Metabase will show only the data points on the dates that have actual data and will not fill the rest with zeroes.

E.g.: lets say we have a table like this one and today's date is 19th of May, and we would like to do a summary of all values till TODAY
```
SELECT 1 AS value, '05-15-2021' AS date
UNION ALL
SELECT 3 AS value, '05-16-2021' AS date
UNION ALL
SELECT 2 AS value, '05-17-2021' AS date
```
it should show a bar chart like this one
![image](https://user-images.githubusercontent.com/1711649/118860144-6cdbd300-b8b1-11eb-8c92-61e99fcd19ef.png)

currently Metabase will do:
![image](https://user-images.githubusercontent.com/1711649/118860262-909f1900-b8b1-11eb-86bf-f40355d93fd7.png)

as it doesn't have any data in both 18th and 19th

**How to Test This Feature Out**

When using filters that use the current date as the end date, then Metabase should fill the remaining dates with zeroes

**Comment by NicolasPA** [comment]
I have some rows count histograms to see if data was imported for each day, and that would help to find if data is missing for today faster.

In fact, a broader "zero-fill" feature, which would automatically add a 0 for dates where data is missing in the selected date interval, would be the killer feature, fixing this issue and more.  
It's possible to do it with SQL, but it's not convenient.

**Comment by mltsy** [comment]
Yes!  This would be great.  I think a better title for this would be "x-axis should respect selected date range, regardless of missing data".  This causes problems on a Dashboard when comparing charts containing, for instance, weekly data where one chart has no data for Monday - it causes the charts to be confusing and hard to visually compare.

Created a discussion about this here: https://discourse.metabase.com/t/simple-count-graphed-as-a-timeseries-for-previous-6-months-is-missing-dates/35570/3

**Comment by christianvuerings** [comment]
Has anyone found a workaround for this issue?

**Comment by mltsy** [comment]
Not without editing the SQL to using Date math to select data for each date individually (which honestly I haven't even tried, because that's just a big mess to maintain)
